opennms-dao/src/main/java/org/opennms/netmgt/dao/hibernate/CdpElementDaoHibernate.java

```diff
@@ -57,7 +60,12 @@ public CdpElement findByNodeId(Integer id) {
 
 	@Override
 	public CdpElement findByGlobalDeviceId(String deviceId) {
-        return findUnique("from CdpElement rec where rec.cdpGlobalDeviceId = ? ", deviceId);
+        List<CdpElement> elements = find("from CdpElement rec where rec.cdpGlobalDeviceId = ? ", deviceId);
+        if (elements.size() > 1) {
+            LoggerFactory.getLogger(getClass()).warn("Expected one CdpElement for device with id {} but found {}. Using CdpElement {} and ignoring others.", deviceId, elements.size(), elements.get(0));
+        }
+        return elements.isEmpty() ? null : elements.get(0);
```

Comment:
I think we should make sure that the list of sorted by some criteria, so that we consistently choose the same element if/when there are many.